pandas/core/indexes/base.py

```diff
@@ -2263,21 +2263,37 @@ def __or__(self, other):
     def __xor__(self, other):
         return self.symmetric_difference(other)
 
-    def _get_consensus_name(self, other):
+    def _get_reconciled_name_object(self, other):
         """
-        Given 2 indexes, give a consensus name meaning
-        we take the not None one, or None if the names differ.
-        Return a new object if we are resetting the name
+        If the result of a set operation will be self,
+        return self, unless the name changes, in which
+        case make a shallow copy of self.
         """
-        if self.name != other.name:
-            if self.name is None or other.name is None:
-                name = self.name or other.name
-            else:
-                name = None
-            if self.name != name:
-                return self._shallow_copy(name=name)
+        name = get_op_result_name(self, other)
+        if self.name != name:
+            return self._shallow_copy(name=name)
         return self
 
+    def _union_corner_case(self, other):
```

Comment:
is there a reason you are creatijng this as a separate method? why not inline it

Reply:
+1 for inlining it.

Reply:
@jreback @TomAugspurger The subclasses for `DateTimeIndex`, `RangeIndex` and `TimeDeltaIndex` need to call this method from their `union` implementations.  Should I inline it there too and have repeated code?

Reply:
ok this PR looks a lot better. I still don't like this hlper function. can we refactor to not use it